src/Symfony/Component/Serializer/Debug/TraceableSerializer.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Debug;
+
+use Symfony\Component\Serializer\SerializerInterface;
+use Symfony\Contracts\Service\ResetInterface;
+
+final class TraceableSerializer implements SerializerInterface, ResetInterface
+{
+    /**
+     * @var Serialization[]
+     */
+    private $serializations = [];
+    /**
+     * @var Deserialization[]
+     */
+    private $deserializations = [];
+
+    /**
+     * @var SerializerInterface
+     */
+    private $delegate;
+
+    public function __construct(SerializerInterface $delegate)
+    {
+        $this->delegate = $delegate;
+    }
+
+    public function serialize($data, string $format, array $context = []): string
+    {
+        $serialization = new Serialization($data, $format, $context);
```

Comment:
I'm not sure we need an object to store these information.
an `array` could be enough IMHO

Reply:
I'm not a fan of arrays for structured data and neither should anyone. I'd like to stick with this (somehow) struct in here.